### PR TITLE
MANTA-5001 want tests for bucket and object name validation in node-manta

### PR DESCRIPTION
**Pull request by bahamas10**

### CODE REVIEW

File: test/integration/buckets-client-validation.test.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    'Uppercase-Letters',
+
+    'nul\u0000byte',
+
+    // 2 characters (too short)
+    'ab',
+
+    // 64 characters (too long)
+    'abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijkl'
+];
```
Comment (trentm) [inline test/integration/buckets-client-validation.test.js]: Can we add some non-ascii test cases? I assume those are disallowed?

File: test/integration/buckets-client-validation.test.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    'bar',
+    maxBytesString
+];
+
+// 1024 bytes max
+
+var INVALID_OBJECT_NAMES = [
+    'nul\u0000byte',
+    maxBytesString + 'a'
+];
```
Comment (trentm) [inline test/integration/buckets-client-validation.test.js]: Can we have a unicode test that ensures that a unicode-containing string that is less than 1024 chars, but greater than 1024 *bytes* as UTF-8 is disallowed?

File: test/integration/buckets-client-validation.test.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        t.end();
+    });
+
+    test('isBucketsSupported', function (t) {
+        client.isBucketsSupported(function (err, isSupported) {
+            t.ifError(err);
+            t.ok(isSupported, 'isSupported');
+            t.end();
+        });
+    });
```
Comment (trentm) [inline test/integration/buckets-client-validation.test.js]: Is this needed?  `testOpts` above should handle guarding against this already.

File: test/integration/buckets-client-validation.test.js (hunk 1 above)
Comment (bahamas10) [reply test/integration/buckets-client-validation.test.js]: will do

File: test/integration/buckets-client-validation.test.js (hunk 2 above)
Comment (bahamas10) [reply test/integration/buckets-client-validation.test.js]: yep, will do.

File: test/integration/buckets-client-validation.test.js (hunk 3 above)
Comment (bahamas10) [reply test/integration/buckets-client-validation.test.js]: probably not... i can remove this.

**Pull request merged**